modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/CopyTreesFromDB.pm

```diff
@@ -278,15 +253,24 @@ sub write_output {
 ##########################################
 
 # This function is used to remap the old seq_member_ids in the re_used tree with the new ones used in the current_tree.
-# This is necessary for the cases where a species have been updated (gene-set || assembly), this means that the seq_members will be
-#   re-inserted therefore not sharing the same ids with the reusede database.
+# This is necessary for the cases where a species has been updated (gene-set || assembly).
+# This means that the seq_members will be re-inserted in order to avoid conflict with reuse db.
 # For more details check Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::FlagUpdateClusters::_seq_member_map
 sub _remap_leaves {
     my ( $self, $all_leaves ) = @_;
 
     foreach my $leaf ( @{$all_leaves} ) {
         my $stable_id = $leaf->stable_id;
-        my $sql = "SELECT seq_member_id_current FROM seq_member_id_current_reused_map WHERE stable_id = '$stable_id'";
+        my $old_member_id = $leaf->dbID;
+        my $sql = qq/
+        SELECT
+            seq_member_id_current
+        FROM
+            seq_member_id_current_reused_map
+        WHERE
+            stable_id = "$stable_id"
+                AND seq_member_id_reused = $old_member_id;
+            /;
```

Comment:
Is it a common approach to include SQL code like this here? Is there any specified layer for querying the DB?

Reply:
In this case it is because this is an ad-hoc(ish) ehive pipeline specific table in the pipeline database, there is no API/adaptor for this table, so it has to be done like this

Reply:
Ah I see - so Is an adaptor need to be created via a ticket?

Reply:
No, it wouldn't be worth the development, it is purely for an ehive specific process/analysis used in production and should not be part of the Compara API - one of the sorts of things that needs to be considered as we move away from Perl and eHive :grimacing: